zulipterminal/ui_tools/views.py

```diff
@@ -934,36 +934,54 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
 PopUpViewTableContent = Sequence[Tuple[str, Sequence[Union[str, Tuple[str, str]]]]]
 
 
-class PopUpView(urwid.ListBox):
+class PopUpView(urwid.Frame):
     def __init__(
         self,
         controller: Any,
-        widgets: List[Any],
+        body: List[Any],
         command: str,
         requested_width: int,
         title: str,
+        header: Optional[List[Any]] = None,
+        footer: Optional[List[Any]] = None,
     ) -> None:
         self.controller = controller
         self.command = command
         self.title = title
-        self.log = urwid.SimpleFocusListWalker(widgets)
+        self.log = urwid.ListBox(urwid.SimpleFocusListWalker(body))
 
         max_cols, max_rows = controller.maximum_popup_dimensions()
 
         self.width = min(max_cols, requested_width)
 
-        height = self.calculate_popup_height(widgets, self.width)
+        height = self.calculate_popup_height(body, header, footer, self.width)
         self.height = min(max_rows, height)
+        if header:
+            header = urwid.Pile(urwid.SimpleFocusListWalker(header))
```

Comment:
You are absolutely correct, but since this is in `PopupView` which is used in multiple places right now, I wanted to have a generic solution that doesn't need much tinkering to make it work for slightly different use-cases as we had to do in this PR.

We might want to have a list of `LineBox`(or some other widget) as header in the future, then this approach would not need amendments and would work for both cases. Let me know if you still want me to change my approach. :)

Reply:
We could still make that work but still use simpler widgets. The PopupView could accept a single widget and wrap it inside a LineBox and send it to Frame. In your case, you can directly send the PanelSearchBox. For other new Popups, mostly headers would contain 1 widget but if they really want to use more than that, they could send a Pile. Anything can be inside Pile: Walker, LineBox, etc.
Another issue in your version is that I have no way of manipulating the final "log" (inside Pile) if I want to. I would have to create another log in the child which would not be a nice way to do things.

Reply:
Can I use `self.header._w` or something here? Anyway having a list in the header would be an exception and it is better if it was made and handled locally in the child class.
Also every child using the header will for sure use a LineBox. So instead of each child making their own LineBox, its better if PopupView does it once?

What do others think here?

Reply:
I am keeping this unchanged for now, at least until we get more feedback on this. Nonetheless, this is definitely a change worth considering/discussing. :+1: